src/azure-cli/azure/cli/command_modules/iot/tests/latest/test_iot_dps_commands.py

```diff
@@ -43,12 +46,15 @@ def test_dps_lifecycle(self, group_name, group_location):
 
         property_to_update = 'properties.allocationPolicy'
         updated_value = 'GeoLatency'
+        updated_tags = "key3=value3"
         # Update DPS
-        self.cmd('az iot dps update -g {} -n {} --set {}="{}"'.format(group_name, dps_name, property_to_update, updated_value), checks=[
-            self.check('name', dps_name),
-            self.check('location', group_location),
-            self.check(property_to_update, updated_value)
-        ])
+        updated_dps = self.cmd('az iot dps update -g {} -n {} --tags {} --set {}="{}"'
+                               .format(group_name, dps_name, updated_tags, property_to_update, updated_value),
+                               checks=[self.check('name', dps_name),
+                                       self.check('location', group_location),
+                                       self.check(property_to_update, updated_value)]).get_output_in_json()
+
+        assert len(updated_dps['tags']) == 3
```

Comment:
Is updating tags an incremental operation rather than a full coverage operation?

Reply:
Yes, Update is an incremental operation

Reply:
For other RP, however, modifying the tags is generally a full coverage operation. Is it necessary to consider consistency with other RPS?
What's more, if a customer wants to delete some tags, what should he do?

Reply:
Let's add a test which covers removing tags as well.

Reply:
Added a scenario for deleting the tags

Reply:
Thank you @anusapan. I think @zhoxing-ms is suggesting most other RP's CLI wise are a full coverage operation for the `--tags` argument. To support a consistent experience for users (at least for common shared attributes) it makes sense for us to go in this direction as well.

Reply:
According to the help content for parameter "--tags", use "" to delete the existing tags.
![image](https://user-images.githubusercontent.com/31940305/98187135-14b92280-1ec5-11eb-9743-fcc731d030e0.png)

Reply:
Yea for sure. But my comment is not referring to the delete convention specifically. It's referring to how the current state of this PR supports an incremental update to the tags dictionary `parameters.tags.update(tags)` where-as most other RP's in the CLI don't work this way (zhoxing-ms comment earlier). So to be consistent we should assign the tags attribute every time. If the input is `""` the tags param changes the value to `{}` thereby resetting existing tags.

Reply:
Updated the PR, Now there is no incremental update for the tags, if passing tags parameter value then it will overwrite the existing tags.